### ISSUE TITLE
False positive for LICENSE file

### ISSUE THREAD

**Issue by jessevdp** [sindresorhus/awesome-lint#91]
In an effort to remove all linting errors from the [awesome-jamstack](https://github.com/automata/awesome-jamstack) repository...

Awesome-lint is showing an error regarding the fact that there is no LICENSE for the [awesome-jamstack](https://github.com/automata/awesome-jamstack) repository. The project already has a license, which is linked to from the README, but this is not detected. In an effort to resolve this issue I've copied their license from its source and added it in a `LICENSE` file in the root of the repository. And opened a PR: https://github.com/automata/awesome-jamstack/pull/33

Unfortunately, even after deliberately adding the license in a separate file, the build for that repository still fails because the license is not detected. You can check out the build on any commit in that PR, or see [here](https://travis-ci.org/automata/awesome-jamstack/builds/593407519?utm_source=github_status&utm_medium=notification).

---

How does the linter scan for the LICENSE? And how can I make it detect it? Or is this a bug?

**Comment by jessevdp** [comment]
The error mentions that "the license is not detected by ***GitHub***. This leads me to believe that MAYBE you scan for the license on the default branch? That would seem odd to me, as a linter should look at the source of the current branch, right?

**Comment by nschloe** [comment]
Bumped into the same thing with https://github.com/nschloe/awesome-scientific-computing.

I now just added a CC0 file, pushed it, and activated awesome-lint for tests in a second PR.

**Comment by jessevdp** [comment]
@nschloe—so you confirm it only detects the license after it has been pushed to the default branch?

**Comment by nschloe** [comment]
@jessevdp Yes indeed. Not that there was any doubt about that. I mean, how else is awesome-lint going to do it? First, you'll _have_ to push a license to the default branch, GitHub can then check it, and only _then_ can the linter do anything.

**Comment by jessevdp** [comment]
So I took a look at the source to figure out what is happening. The linting error comes from [`rules/github.js`](https://github.com/sindresorhus/awesome-lint/blob/master/rules/github.js). This rule uses the git CLI to get a URL for the remote origin, parses this URL, retrieves the API endpoint for the repository, and then fetches the repo information from that API endpoint.

From looking at that return value of an API endpoint I concluded that (as @nschloe mentioned) GitHub checks the license on the repository level, not on the branch level. I don't know if there is any way to check this on branch level.

The workaround that @nschloe suggested should work fine. Figuring out how to check the license depending on the branch is probably not worth the effort.
